### PR TITLE
Show enrolled extra curricular courses in the Student portal ACDM-663 #resolve

### PR DESCRIPTION
**Pull request by ccostinha91**
Show enrolled extra curricular courses in the Student portal after cycle is concluded

### CODE REVIEW

File: src/main/java/org/fenixedu/academic/ui/struts/action/student/ShowStudentPortalDA.java [hunk 1]
```diff
@@ -62,24 +63,26 @@
...
-                if (registration.getAttendingExecutionCoursesForCurrentExecutionPeriod().isEmpty() == false) {
-                    studentPortalBeans.add(new StudentPortalBean(registration.getDegree(), student, registration
-                            .getAttendingExecutionCoursesForCurrentExecutionPeriod(), degreeCurricularPlan));
-                }
-                if (hasSpecialSeasonEnrolments(student)) {
-                    genericDegreeWarnings.addAll(getEnrolmentPeriodCoursesAfterSpecialSeason(degreeCurricularPlan));
-                } else {
-                    genericDegreeWarnings.addAll(getEnrolmentPeriodCourses(degreeCurricularPlan));
+            for (final Registration registration : student.getAllRegistrations()) {
+                if (considerRegistration(registration.getLastStateType())) {
```
Comment (ricardorcr) [inline src/main/java/org/fenixedu/academic/ui/struts/action/student/ShowStudentPortalDA.java]: This filter is not necessary, the way to validate "acceptable" registrations for this is way to specific and is subject to get outdated with more states being added eventually.
The check is done in each registration for attends.

**Comment by ricardorcr** [comment]
Please keep the change set to the minimum and to what was really necessary to do for the issue.
Reformating the code and adding "final" attributes and such in not new code makes it harder to see what real was changed.

**Comment by ccostinha91** [comment]
Eclipse made that changes automatically when saving the document. Should I change it?

**Comment by ricardorcr** [comment]
Yes, change that and leave it as it was.

**Comment by ccostinha91** [comment]
its done

**Comment by Luis-Cruz** [comment]
Merged manually.

**Pull request closed without merging**